packages/core/src/components/graphs/alluvial.ts

```diff
@@ -0,0 +1,596 @@
+					event,
+					element: select(this),
+					datum,
+				});
+			})
+			.on('mouseout', function (event, datum) {
+				const hoveredElement = select(this);
+
+				// Set the node position to initial state (unexpanded)
+				const nodeMatrix = Tools.getTranformOffsets(
+					hoveredElement.attr('transform')
+				);
+
+				hoveredElement
+					.classed('hovered', false)
+					.attr(
+						'transform',
+						`translate(${nodeMatrix.x + 2}, ${nodeMatrix.y})`
+					)
+					.select('rect.node')
+					.attr('width', Configuration.alluvial.nodeWidth);
+
+				// Translate text container back to initial state
+				if (datum.x0 - 2 === 0) {
+					const titleContainer = self.parent.select(
+						`g#node-title-${datum.index}`
+					);
+					const titleMatrix = Tools.getTranformOffsets(
+						titleContainer.attr('transform')
+					);
+
+					titleContainer.attr(
+						'transform',
+						`translate(${titleMatrix.x - 4},${titleMatrix.y})`
+					);
+				}
+
+				self.parent
+					.select(`text#node-text-${datum.index}`)
+					.style('font-weight', 'normal');
+
+				debouncedLineHighlight([], 'mouseout');
+
+				// Dispatch mouse out event
+				self.services.events.dispatchEvent(
+					Events.Alluvial.NODE_MOUSEOUT,
+					{
+						event,
+						element: hoveredElement,
+						datum,
+					}
+				);
+
+				// Dispatch hide tooltip event
+				self.services.events.dispatchEvent(Events.Tooltip.HIDE, {
+					hoveredElement,
+				});
+			});
+	}
+
+	// Sets the opacity of all lines to default (0.8)
+	private normalizeLinks(nodes = false) {
+		const links = this.parent
+			.selectAll('path.link')
+			.classed('hovered', false)
+			.style('stroke-opacity', Configuration.alluvial.opacity.default);
+
+		// Correct link orders if normalizing from node hover
+		if (nodes) {
+			links.data(this.graph.links, (d) => d.index).order();
+		}
+	}
+
+	// Sets the opacity of all lines to unfocused (0.3)
+	private unhighlightLines() {
+		this.parent
+			.selectAll('path.link')
+			.style('stroke-opacity', Configuration.alluvial.opacity.unfocus);
+	}
+
+	// Traverse graph and get all connected links to node
+	private traverse(
+		direction:
+			| { link: 'sourceLinks'; node: 'target' }
+			| { link: 'targetLinks'; node: 'source' },
+		node,
+		visited = []
+	) {
+		const links = node[direction.link].map((element) => {
+			visited.push(`path#line-${element.index}.link`);
+			return element[direction.node];
+		});
+
+		// Retrieve the child nodes
+		links.forEach((element) => this.traverse(direction, element, visited));
+	}
+
+	getRightArrowIcon() {
+		return `
+		<svg id="icon" xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 32 32">
+		<defs>
+			<style>
+			.cls-1 {
+				fill: none;
+			}
+			</style>
+		</defs>
+		<polygon points="18 6 16.57 7.393 24.15 15 4 15 4 17 24.15 17 16.57 24.573 18 26 28 16 18 6"/>
+		<rect id="_Transparent_Rectangle_" data-name="&lt;Transparent Rectangle&gt;" class="cls-1" width="32" height="32"/>
+		</svg>`;
```

Comment:
could you pls clean up this svg? we can remove the classname & just add the style as an attribute to the `rect`, and the `id` seems redundant bc. if we have multiple, then it's a problem